### ISSUE TITLE
UWP + EFC, cross-target frameworks ignored

### ISSUE THREAD

**Issue by BenjaVR** [aspnet/EntityFrameworkCore#10882]
When I make a UWP app, and a .NET Standard class library with Entity Framework Core, I should use cross-target like this:
`<TargetFrameworks>netcoreapp2.0;netstandard2.0</TargetFrameworks>`

Although, when I put the 'netstandard2.0' first, my EF will not add a new migration. And if I put the 'netcoreapp2.0' first, Visual Studio will not be able to reference this class library from my UWP project.

### Steps to reproduce

1. Create a new empty UWP project, targeting Fall Creators Update.
![image](https://user-images.githubusercontent.com/17981138/35831499-33af3754-0aca-11e8-9167-028f515f04ab.png)
![image](https://user-images.githubusercontent.com/17981138/35831534-50537f32-0aca-11e8-801c-9f37490d932f.png)

2. In the solution, create a new project: .NET Standard class library.
![image](https://user-images.githubusercontent.com/17981138/35831636-b7abf240-0aca-11e8-9247-63084b66a65c.png)

3. Add a reference from the class library to the UWP project.
![image](https://user-images.githubusercontent.com/17981138/35831735-0ea3f8ea-0acb-11e8-99b7-9e9eb0e8c038.png)

4. Add EF Nuget packages to the class library.
![image](https://user-images.githubusercontent.com/17981138/35831882-86c46292-0acb-11e8-8513-cfdb10ce545f.png)

5. Make a model and DbContext.
```c#
using System.ComponentModel.DataAnnotations;

namespace UWP_EF_TEST.Data
{
    public class Item
    {
        [Key]
        public int Id { get; set; }

        [Required]
        public string Value { get; set; }
    }
}
```
```c#
using Microsoft.EntityFrameworkCore;

namespace UWP_EF_TEST.Data
{
    public class MyDbContext : DbContext
    {
        public DbSet<Item> Items { get; set; }

        protected override void OnConfiguring(DbContextOptionsBuilder optionsBuilder)
        {
            optionsBuilder.UseSqlite("test.db");
        }
    }
}
```
![image](https://user-images.githubusercontent.com/17981138/35832443-a1dd891c-0acd-11e8-9baa-0eb5cd3fb8c6.png)

6. Try to run `Add-Migration Test` in the PMC, after setting the startup project to your class library. This will fail.
![image](https://user-images.githubusercontent.com/17981138/35832544-f70d99b8-0acd-11e8-9d29-252112452119.png)
Error:
`
Startup project 'UWP_EF_TEST.Data' targets framework '.NETStandard'. There is no runtime associated with this framework, and projects targeting it cannot be executed directly. To use the Entity Framework Core Package Manager Console Tools with this project, add an executable project targeting .NET Framework or .NET Core that references this project, and set it as the startup project; or, update this project to cross-target .NET Framework or .NET Core.
`

7. Setup cross-target frameworks in the .csproj of your class library. Also make sure to generate migrations by adding another extra line.
![image](https://user-images.githubusercontent.com/17981138/35833081-f815f1e6-0acf-11e8-9443-be692d7107a4.png)
```xaml
<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <TargetFrameworks>netstandard2.0;netcoreapp2.0</TargetFrameworks>
    <GenerateRuntimeConfigurationFiles>true</GenerateRuntimeConfigurationFiles>
  </PropertyGroup>

  <ItemGroup>
    <PackageReference Include="Microsoft.EntityFrameworkCore" Version="2.0.1" />
    <PackageReference Include="Microsoft.EntityFrameworkCore.Sqlite" Version="2.0.1" />
    <PackageReference Include="Microsoft.EntityFrameworkCore.Tools" Version="2.0.1" />
  </ItemGroup>

</Project>
```

8. Try `Add-Migration Test` in the PMC again. It will fail again, with the exact same message.
![image](https://user-images.githubusercontent.com/17981138/35833136-311e07f8-0ad0-11e8-8c26-d1d0aa747fc6.png)

9. Switch the TargetFrameworks.
![image](https://user-images.githubusercontent.com/17981138/35833180-6a910922-0ad0-11e8-8bf9-ebd4909ae078.png)

10. Third try to run the `Add-Migration Test`.
![image](https://user-images.githubusercontent.com/17981138/35833452-9d59b1a0-0ad1-11e8-8c99-158085a34131.png)

11. Cool, it worked! Let's close Visual Studio, and restart it. Now it suddenly can't add the reference anymore... Everything works fine (I can run whatever app I make), but the error won't go away. Even if I remove the reference, it's not possible to add it again. I have to switch the `netcoreapp2.0` and `netstandard2.0` again to make the reference. So the multiple target kind of gets ignored.
![image](https://user-images.githubusercontent.com/17981138/35833532-f4035d1c-0ad1-11e8-9226-d06253643a37.png)

---
Please note, I'm not sure if this is an issue for EFCore, or Visual Studio, or some other project.

### Further technical details
EF Core version: 2.0.1
Database Provider: Microsoft.EntityFrameworkCore.Sqlite
Operating system: Windows 10 Pro, version 1709, build 16299.192
IDE: Visual Studio 2017 15.4

**Comment by bricelam** [comment]
Correct. The Visual Studio APIs return values for the first framework listed. I'm unaware of a way to specify which one to use.

**Comment by BenjaVR** [comment]
Hmm, and do you know of any way to use UWP + EF without any hassles and errors etc?

**Comment by bricelam** [comment]
We could follow up with [dotnet/project-system](https://github.com/dotnet/project-system).

The `dotnet ef` command has the same behavior; we could follow up with [dotnet/sdk](https://github.com/dotnet/sdk) for possible solutions there.

**Comment by bricelam** [comment]
Many people have created a dummy .NET Core console app to use with the tools. Specify `-StartupProject TheConsoleApp`.

**Comment by BenjaVR** [comment]
Will try that one, thanks!

Do you want me to post this issue in those repositories?

**Comment by bricelam** [comment]
#9666 is about making the UWP experience better

**Comment by bricelam** [comment]
> Do you want me to post this issue in those repositories?

No, we'll follow up. I have a feeling we can better articulate exactly what we need. 😉

**Comment by BenjaVR** [comment]
Sure, no doubt about it, thanks!

**Comment by bricelam** [comment]
Notes
===
In `dotnet ef`, we could probably use the `GetNearestTargetFramework` task to be smarter.

**Comment by bricelam** [comment]
We followed up with the project system team. They have no plans to change the behavior of the Visual Studio APIs (EnvDTE), nor to add first-class UI for changing the active framework.

Resolving this will require us to use lower level APIs to understand cross-targeting projects.

A first step would be to match `dotnet ef` and add a `-Framework` parameter.

After that, we could look into making `dotnet ef` and the PMC tools smarter about which framework it used by default.

**Comment by ajcvickers** [comment]
Putting this on the backlog now, to reconsider based on feedback. Also, [updated the docs](https://docs.microsoft.com/en-us/ef/core/miscellaneous/cli/) to not recommend cross-targeting.
